Core/Utilities/ThreadPool.py

```diff
@@ -310,9 +314,22 @@ def getGlobalThreadPool():
     gThreadPool.daemonize()
   return gThreadPool
 
+gThreadPoolExecutor = False
+def getGlobalThreadPoolExecutor():
+  global gThreadPoolExecutor
+  if not gThreadPoolExecutor:
+    gThreadPoolExecutor = ThreadPoolExecutor( 500 )
+  return gThreadPoolExecutor
```

Comment:
to make sure that we are not using n thread pool and make sure that we will not run out the number of threads.